src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        firstNode = null;
+        lastNode = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            firstNode = newNode;
+            lastNode = newNode;
+        } else {
+            newNode.prev = lastNode;
+            lastNode.next = newNode;
+            lastNode = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            newNode.next = firstNode;
+            firstNode.prev = newNode;
+            firstNode = newNode;
+            size++;
+        } else {
+            addNodeInside(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> result = getNodebyIndex(index);
+        return result.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> result = getNodebyIndex(index);
+        T oldVal = result.value;
```

Comment:
please try to not cut the words in variables  names
```suggestion
        T oldValue = result.value;
```